pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1058,13 +1086,44 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListenerForPortOrUDS(l
 		}
 
 		listenerOpts.filterChainOpts = opts
+
 	case plugin.ListenerProtocolTCP:
 		if ret, opts = configgen.buildSidecarOutboundTCPListenerOptsForPortOrUDS(&destinationCIDR, &listenerMapKey, &currentListenerEntry,
 			&listenerOpts, pluginParams, listenerMap, virtualServices, actualWildcard); !ret {
 			return
 		}
 
 		listenerOpts.filterChainOpts = opts
+
+	case plugin.ListenerProtocolAuto:
+		if !util.IsProxyVersionGE13(node) {
+			return
+		}
+
+		if ret, opts = configgen.buildSidecarOutboundHTTPListenerOptsForPortOrUDS(&listenerMapKey, &currentListenerEntry,
```

Comment:
Same as my previous comment ...  add comments throughout this section to explain how it's configured differently than other protocols.